src/utilities/image-utils.ts

```diff
@@ -108,3 +109,21 @@ export function getImageDimensions(callback: any, file?: File, url?: string) {
       image.src = url!;
     }
 }
+
+// new Promise-based way to get dimensions
+export function getDimensions(file?: File, url?: string): Promise<any> {
```

Comment:
Can we get rid of the callback-based `getImageDimensions()` function and call this one `getImageDimensions()`?
```suggestion
export function getImageDimensions(url: string): Promise<any> {
```
We never call this with a file, so let's simplify the API. If we need to support files at some point, we could add another function like:
```typescript
export function getImageFileDimensions(file: File) {
  return getDimensions(URL.createObjectURL(file));
}
```
Can we get rid of